core/src/main/java/org/mskcc/cbio/portal/dao/DaoGenericAssay.java

```diff
@@ -85,4 +85,29 @@ public static void deleteGenericEntityPropertiesByStableId(String stableId) thro
             JdbcUtil.closeAll(DaoGeneticEntity.class, con, pstmt, rs);
         }
     }
+
+    public static boolean shouldGenericAssayMetaBeDeleted(int cancerStudyId) throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+
+        try {
+            con = JdbcUtil.getDbConnection(DaoGeneticEntity.class);
+            pstmt = con.prepareStatement("SELECT DISTINCT CANCER_STUDY_ID FROM genetic_profile WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_alteration WHERE GENETIC_ENTITY_ID IN (SELECT GENETIC_ENTITY_ID FROM genetic_alteration WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_profile WHERE CANCER_STUDY_ID=?)))");
```

Comment:
Seems a bit circular, but the true test is below, that the result set size == 1.